src/resolver.js

```diff
@@ -76,26 +76,34 @@ function resolverFactory(targetMaybeThunk, options = {}) {
         findOptions.order = [[model.primaryKeyAttribute, 'ASC']];
       }
 
+      function checkConnection() {
```

Comment:
i like this approach but defining the functions everytime seems like a waste, move them outside and have them take all args required please.